src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -120,7 +125,13 @@ - (instancetype)initWithFactory:(MTRDeviceControllerFactory *)factory queue:(dis
             return nil;
         }
         _operationalCredentialsDelegate->setChipWorkQueue(_chipWorkQueue);
+
+        _deviceDiscoveryDelegateBridge = new MTRDeviceDiscoveryDelegateBridge();
```

Comment:
If this is going to get checked for null, doesn't the `new` need to be `std::nothrow`?

Reply:
I removed all discovery and ethernet pairing changes because they are useful for test but not a mandatory feature for Matter certification.